index.js

```diff
@@ -1,181 +1,159 @@
-    type: 'boolean',
-    default: false,
-    global: true
-  })
-  .option('sign', {
-    alias: 's',
-    describe: 'Should the git commit and tag be signed?',
-    type: 'boolean',
-    default: false,
-    global: true
-  })
-  .option('no-verify', {
-    alias: 'n',
-    describe: 'Bypass pre-commit or commit-msg git hooks during the commit phase',
-    type: 'boolean',
-    default: false,
-    global: true
-  })
-  .help()
-  .alias('help', 'h')
-  .example('$0', 'Update changelog and tag release')
-  .example('$0 -m "%s: see changelog for details"', 'Update changelog and tag release with custom commit message')
-  .wrap(97)
-  .argv
 
 var chalk = require('chalk')
 var figures = require('figures')
 var exec = require('child_process').exec
 var fs = require('fs')
 var accessSync = require('fs-access').sync
-var pkgPath = path.resolve(process.cwd(), './package.json')
-var pkg = require(pkgPath)
 var semver = require('semver')
 var util = require('util')
+var objectAssign = require('object-assign')
 
-conventionalRecommendedBump({
-  preset: 'angular'
-}, function (err, release) {
-  if (err) {
-    console.error(chalk.red(err.message))
-    return
-  }
+module.exports = function standardVersion (argv, done) {
+  var pkgPath = path.resolve(process.cwd(), './package.json')
+  var pkg = require(pkgPath)
+  var defaults = require('./defaults')
 
-  var newVersion = pkg.version
-  if (!argv.firstRelease) {
-    newVersion = semver.inc(pkg.version, release.releaseType)
-    checkpoint('bumping version in package.json from %s to %s', [pkg.version, newVersion])
-    pkg.version = newVersion
-    fs.writeFileSync(pkgPath, JSON.stringify(pkg, null, 2) + '\n', 'utf-8')
-  } else {
-    checkpoint('skip version bump on first release', [], chalk.red(figures.cross))
-  }
+  argv = objectAssign(defaults, argv)
 
-  outputChangelog(argv, function () {
-    commit(argv, newVersion, function () {
-      return tag(newVersion, argv)
-    })
-  })
-})
-
-function outputChangelog (argv, cb) {
-  createIfMissing(argv)
-  var header = '# Change Log\n\nAll notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.\n'
-  var oldContent = fs.readFileSync(argv.infile, 'utf-8')
-  // find the position of the last release and remove header:
-  if (oldContent.indexOf('<a name=') !== -1) {
-    oldContent = oldContent.substring(oldContent.indexOf('<a name='))
-  }
-  var content = ''
-  var changelogStream = conventionalChangelog({
+  conventionalRecommendedBump({
     preset: 'angular'
-  })
-  .on('error', function (err) {
-    console.error(chalk.red(err.message))
-    process.exit(1)
-  })
+  }, function (err, release) {
+    if (err) {
+      printError(err.message)
+      return done(err)
+    }
 
-  changelogStream.on('data', function (buffer) {
-    content += buffer.toString()
-  })
+    var newVersion = pkg.version
```

Comment:
Yeah I first structured it like that, but will have to make sure everything is passed around correctly. I'll try to work that in before landing this.